qml/Main.qml

```diff
@@ -26,7 +26,90 @@ ApplicationWindow
+    property Page _mainPage
+
+    allowedOrientations: defaultAllowedOrientations
+    _defaultLabelFormat: Text.PlainText
+    _defaultPageOrientations: Orientation.All
+    cover: Qt.resolvedUrl("CoverPage.qml")
+    initialPage: Component {
+        FileListPage {
+            id: fileListPage
+
+            model: trackerProvider.model
+            provider: trackerProvider
+            Component.onCompleted: window._mainPage = fileListPage
+        }
+    }
+
+    /* Currently tracker is the only source for documents, so DocumentProviderListModel is unused
+    DocumentProviderListModel {
+        id: documentProviderListModel
+
+    }*/
+    TrackerDocumentProvider {
+        id: trackerProvider
+    }
+
+    FileInfo {
+        id: fileInfo
+    }
+
+    function openFile(file) {
+        fileInfo.source = file
+
+        pageStack.pop(window._mainPage, PageStackAction.Immediate)
+
+        var handler = ""
+
+        switch (fileInfo.mimeType) {
+        case "application/vnd.oasis.opendocument.spreadsheet":
+        case "application/x-kspread":
+        case "application/vnd.ms-excel":
+        case "text/csv":
+        case "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet":
+        case "application/vnd.openxmlformats-officedocument.spreadsheetml.template":
+            handler = "Sailfish.Office.SpreadsheetPage"
+            break
+
+        case "application/vnd.oasis.opendocument.presentation":
+        case "application/vnd.oasis.opendocument.presentation-template":
+        case "application/x-kpresenter":
+        case "application/vnd.ms-powerpoint":
+        case "application/vnd.openxmlformats-officedocument.presentationml.presentation":
+        case "application/vnd.openxmlformats-officedocument.presentationml.template":
+            handler = "Sailfish.Office.PresentationPage"
+            break
+
+        case "application/vnd.oasis.opendocument.text-master":
+        case "application/vnd.oasis.opendocument.text":
+        case "application/vnd.oasis.opendocument.text-template":
+        case "application/msword":
+        case "application/rtf":
+        case "application/x-mswrite":
+        case "application/vnd.openxmlformats-officedocument.wordprocessingml.document":
+        case "application/vnd.openxmlformats-officedocument.wordprocessingml.template":
+        case "application/vnd.ms-works":
+            handler = "Sailfish.Office.TextDocumentPage"
+            break
+
+        case "application/pdf":
+            handler = "Sailfish.Office.PDFDocumentPage"
+            break
+
+        default:
+            console.log("Warning: Unrecognised file type for file " + fileInfo.fullPath)
+        }
+
+        if (handler != "") {
+            pageStack.push(handler,
+                           { title: fileInfo.fileName, path: fileInfo.fullPath, mimeType: fileInfo.mimeType },
+                           PageStackAction.Immediate)
+        }
```

Comment:
What happens if there is no handler? Maybe something should be shown for the user to indicate that this file type is unknown.